src/util.rs

```diff
@@ -147,6 +147,11 @@ pub fn reconstruct_shares(share1: &[Field], share2: &[Field]) -> Option<Vec<Fiel
     Some(reconstructed)
 }
 
+/// Compute the ceiling of the base-2 logarithm of `x`.
+pub fn log2(x: u128) -> u128 {
```

Comment:
>     1. I'd prefer to avoid adding new code to the `util` module (and over time would like to refactor its existing contents into other more appropriate modules), on the premise that `util` modules or `Utilities` classes are generally a code smell. In this case, could we put `log2` in `fft.rs`, since that is where the only non-test use of it resides?
I put it in `fp.rs`.

>     2. All current callsites of `log2` end up casting the returned value to `usize`. Could `log2` return `usize` and save the callers a little trouble?
Line 254 in fp.rs: `let num_roots = log2(order) as usize;` Here, `order` is a `u128` and its value may be in excess of 2^64.
 
>     3. Can the function be `pub(crate)`? n.b. I'm aware that `libprio-rs` rather promiscuously exports symbols, but we ought Done.